arangod/Aql/ExecutionBlockImpl.cpp

```diff
@@ -1969,6 +1980,72 @@ void ExecutionBlockImpl<Executor>::PrefetchTask::execute(ExecutionBlockImpl& blo
   }
 }
 
+template <class Executor>
+ExecutionBlockImpl<Executor>::CallstackSplit::CallstackSplit(ExecutionBlockImpl& block) :
+  _block(block),
+  _thread(&CallstackSplit::run, this, std::cref(ExecContext::current())) {}
+  
+template <class Executor>
+ExecutionBlockImpl<Executor>::CallstackSplit::~CallstackSplit() { 
+  _lock.lock();
+  _state.store(State::Stopped);
+  _lock.unlock();
+
+  _bell.notify_one();
+  _thread.join();
+}
+
+template <class Executor>
+auto ExecutionBlockImpl<Executor>::CallstackSplit::execute(ExecutionContext& ctx, AqlCallType const& aqlCall)
+    -> UpstreamResult {
+  std::variant<UpstreamResult, std::exception_ptr, std::nullopt_t> result{std::nullopt};
+  Params params{result, ctx, aqlCall};
+  
+  _lock.lock();
```

Comment:
I would prefer to use a guard here.
Right now this code is perfectly fine.
But in case something between lock() and unlock() throws the lock will be held, and the Destructor of this class will block as it is trying to get the lock again.

As far as I see everything between lock and unlock is no throw. So for now all is fine, nothing to block this PR for.

Reply:
Done!